pkg/cosign/verify.go

```diff
@@ -721,19 +803,19 @@ func verifyInternal(ctx context.Context, sig oci.Signature, h v1.Hash,
 			return false, err
 		}
 		// If there is no chain annotation present, we preserve the pools set in the CheckOpts.
+		var pool = (*x509.CertPool)(nil)
 		if len(chain) > 0 {
 			if len(chain) == 1 {
 				co.IntermediateCerts = nil
```

Comment:
This line needs to be updated too, we can't overwrite `co.IntermediateCerts`

Reply:
Agreed i will incorporate this comment